sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/RxPartitionKeyRangeCache.java

```diff
@@ -69,7 +70,9 @@ public Mono<Utils.ValueHolder<CollectionRoutingMap>> tryLookupAsync(MetadataDiag
             .onErrorResume(err -> {
                 logger.debug("tryLookupAsync on collectionRid {} encountered failure", collectionRid, err);
                 CosmosException dce = Utils.as(err, CosmosException.class);
-                if (dce != null && Exceptions.isNotFound(dce)) {
+
+                // bubble up in case a 404:1002 is seen to force retries as a part of document retries
+                if (dce != null && Exceptions.isNotFound(dce) && !Exceptions.isSubStatusCode(dce, HttpConstants.SubStatusCodes.READ_SESSION_NOT_AVAILABLE)) {
```

Comment:
hmm, I wonder whether we should really change the condition here: the change here seems indicates SDK can get 404/1002 back while it should not

Reply:
I'll add a todo to revert this change when fault injection excludes 404:1002 for master resources.